### PR TITLE
Remove all usage of deprecated code

### PR DESCRIPTION
**Pull request by provokateurin**

### CODE REVIEW

File: lib/CacheListener.php [hunk 1]
```diff
@@ -9,8 +9,8 @@
 
 use OCA\GroupFolders\Mount\GroupFolderStorage;
 use OCP\EventDispatcher\IEventDispatcher;
-use OCP\Files\Cache\CacheInsertEvent;
-use OCP\Files\Cache\CacheUpdateEvent;
+use OCP\Files\Cache\CacheEntryInsertedEvent;
+use OCP\Files\Cache\CacheEntryUpdatedEvent;
```
Comment (come-nc) [inline lib/CacheListener.php:13]: Do you have a ref for this change in server? Would make for a nice rector.

Comment (provokateurin) [reply lib/CacheListener.php:13]: https://github.com/nextcloud/server/pull/24113

**Review by come-nc — COMMENTED** [review]
Did you test the events in the app to check that they are correctly firing and listened to like before?

**Comment by provokateurin** [comment]
Hehe this even fixes https://github.com/nextcloud/groupfolders/issues/1932 as the old event just wasn't fired anymore it seems.

File: lib/Event/GroupVersionsExpireEnterFolderEvent.php [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\GroupFolders\Event;
+
+use OCP\EventDispatcher\Event;
+
+class GroupVersionsExpireEnterFolderEvent extends Event {
+	public function __construct(
+		public array $folder,
```
[the remark this answers is not in the document]
Comment (provokateurin) [reply lib/Event/GroupVersionsExpireEnterFolderEvent.php:15]: Yeah it's not a Folder Node but the internal representation of a Groupfolder which is just an array.

**Comment by provokateurin** [comment]
> Did you test the events in the app to check that they are correctly firing and listened to like before?

Yes, everything working as expected and for the group deletion event it also fixes a bug :)

**Comment by provokateurin** [comment]
/backport 05c6bcd23d108ae053cae044f2e0699032340c4b to stable30

**Comment by provokateurin** [comment]
/backport 05c6bcd23d108ae053cae044f2e0699032340c4b to stable29

**Comment by provokateurin** [comment]
/backport 05c6bcd23d108ae053cae044f2e0699032340c4b to stable28

**Review by come-nc — APPROVED** [review]

**Pull request merged**